From fb80e4e6c85f562a200970838cd05c357f03c107 Mon Sep 17 00:00:00 2001
From: Amir Tzin <amirtz@nvidia.com>
Date: Mon, 29 May 2023 11:24:19 +0300
Subject: [PATCH 49/58] net/mlx5: Add mlx5_ifc definitions for bridge multicast
 support

Upstream Status: v6.4-rc1

commit e5688f6fb9e3 ("net/mlx5: Add mlx5_ifc definitions for bridge multicast support")
Author: Vlad Buslov <vladbu@nvidia.com>
Date:   Wed Mar 1 10:33:55 2023 +0100

    net/mlx5: Add mlx5_ifc definitions for bridge multicast support

    Add the required hardware definitions to mlx5_ifc: fdb_uplink_hairpin,
    fdb_multi_path_any_table_limit_regc, fdb_multi_path_any_table.

    Signed-off-by: Vlad Buslov <vladbu@nvidia.com>
    Reviewed-by: Maor Dickman <maord@nvidia.com>
    Reviewed-by: Roi Dayan <roid@nvidia.com>
    Signed-off-by: Saeed Mahameed <saeedm@nvidia.com>

Change-Id: I4e1ec56242734b48f441f7d7abccecfc9ce8fa92
Signed-off-by: Amir Tzin <amirtz@nvidia.com>
---
 include/linux/mlx5/mlx5_ifc.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/linux/mlx5/mlx5_ifc.h b/include/linux/mlx5/mlx5_ifc.h
index 03351bc3072b..8ff583d940d3 100644
--- a/include/linux/mlx5/mlx5_ifc.h
+++ b/include/linux/mlx5/mlx5_ifc.h
@@ -815,7 +815,12 @@ enum {
 
 struct mlx5_ifc_flow_table_eswitch_cap_bits {
 	u8      fdb_to_vport_reg_c_id[0x8];
-	u8      reserved_at_8[0xd];
+	u8      reserved_at_8[0x5];
+	u8      fdb_uplink_hairpin[0x1];
+	u8      fdb_multi_path_any_table_limit_regc[0x1];
+	u8      reserved_at_f[0x3];
+	u8      fdb_multi_path_any_table[0x1];
+	u8      reserved_at_13[0x2];
 	u8      fdb_modify_header_fwd_to_table[0x1];
 	u8      fdb_ipv4_ttl_modify[0x1];
 	u8      flow_source[0x1];
-- 
2.31.1

